lib/manageiq/cross_repo/repository.rb

```diff
@@ -46,14 +46,26 @@ def parse_identifier(identifier, server)
         ref  = Dir.chdir(path) { `git rev-parse HEAD`.chomp }
         sha  = ref
       else
-        name, ref = identifier.split("@")
-        ref ||= "master"
+        if identifier.include?("#")
+          name, pr = identifier.split("#")
+        else
+          name, ref = identifier.split("@")
+          ref ||= "master"
+        end
 
         org, repo = name.split("/")
         repo, org = org, "ManageIQ" if repo.nil?
 
         url  = File.join(server, org, repo)
-        sha  = ref.to_s.match?(/^\h+$/) ? ref : `git ls-remote #{url} #{ref}`.split("\t").first
+
+        sha = if pr
+          `git ls-remote #{url} refs/pull/#{pr}/head`.split("\t").first
+        else
+          ref.to_s.match?(/^\h+$/) ? ref : `git ls-remote #{url} #{ref}`.split("\t").first
```

Comment:
Might want to switch this to being part of the if/else directly... like

```ruby
if pr
  git ls...
elsif ref.to_s.match?
  ref
else
  git ls...
end
```

Reply:
Funny you mention that https://github.com/Fryguy/manageiq-cross_repo/pull/15/files#diff-32172a7018d252026b1910784c3806e4R67